pkg/install/hypershift.go

```diff
@@ -205,26 +205,45 @@ func (c *UpgradeController) runHypershiftInstall(ctx context.Context, controller
 		}
 	}
 
-	awsPlatform := true
+	awsPlatform := false
+	oidcBucket := true
+	privateLinkCreds := true
 
+	//Attempt to retrieve oidc bucket secret
 	bucketSecretKey := types.NamespacedName{Name: util.HypershiftBucketSecretName, Namespace: c.clusterName}
 	se := &corev1.Secret{}
 	if err := c.hubClient.Get(ctx, bucketSecretKey, se); err != nil {
-		c.log.Info(fmt.Sprintf("bucket secret(%s) not found on the hub, installing hypershift operator for non-AWS platform.", bucketSecretKey))
+		c.log.Info(fmt.Sprintf("bucket secret(%s) not found on the hub.", bucketSecretKey))
 
-		awsPlatform = false
+		oidcBucket = false
 	}
 
 	// cache the bucket secret for comparison againt the hub's to detect any change
 	c.bucketSecret = *se
 
+	//Attempt to retrieve private link creds
+	privateSecretKey := types.NamespacedName{Name: util.HypershiftPrivateLinkSecretName, Namespace: c.clusterName}
+	spl := &corev1.Secret{}
+	if err := c.hubClient.Get(ctx, privateSecretKey, spl); err != nil {
+		c.log.Info(fmt.Sprintf("private secret(%s) not found on the hub.", privateSecretKey))
+
+		privateLinkCreds = false
+	}
+	// cache the private link secret for comparison againt the hub's to detect any change
+	c.privateLinkSecret = *spl
+
+	//Platform is aws if either secret exists
+	awsPlatform = oidcBucket || privateLinkCreds
```

Comment:
Is `awsPlatform` used anywhere else? The hypershift operator can still be used for AWS platform without any of OIDC or private link secrets. OIDC, private link and externalDNS secrets are just hypershift operator installation options regardless of cloud platform. I am not sure if we really need to detect or say "I am installing the hypershift operator for AWS".

Reply:
It's only used at https://github.com/stolostron/hypershift-addon-operator/blob/22d1b9eb8e49fdf8084966df6b0515ef70a7ccac/pkg/install/hypershift.go#L288-L298 when creating external dns secret